executor/poolmgr/gp.go

```diff
@@ -297,15 +297,28 @@ func (gp *GenericPool) getFetcherUrl(podIP string) string {
 	return fmt.Sprintf("http://%v:8000/", podIP)
 }
 
+func IsIPv6(podIP string) bool {
+	ip := net.ParseIP(podIP)
+	return ip != nil && strings.Contains(podIP, ":")
+}
+
 func (gp *GenericPool) getSpecializeUrl(podIP string, version int) string {
 	u := os.Getenv("TEST_SPECIALIZE_URL")
 	if len(u) != 0 {
 		return u
 	}
+	t := IsIPv6(podIP)
+	if t == false {
+		r := fmt.Sprintf("http://%v:8888", podIP)
```

Comment:
This defines `r` only inside this block, so line 318 below causes a build failure. I think you'll have to define `r` as a var outside the if statement. 

(Also, what does `r` stand for? Maybe use just `baseUrl` or something? And could you also rename `t` to something like `v6`?)